test/e2e/auth/per_node_update.go

```diff
@@ -0,0 +1,231 @@
+		nodeScopedClient, err := kubernetes.NewForConfig(nodeScopedClientConfig)
+		framework.ExpectNoError(err)
+		saUser, err := nodeScopedClient.AuthenticationV1().SelfSubjectReviews().Create(ctx, &authenticationv1.SelfSubjectReview{}, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		expectedUser := serviceaccount.MakeUsername(f.Namespace.Name, "default")
+		o.Expect(saUser.Status.UserInfo.Username).To(o.Equal(expectedUser))
+		expectedNode := authenticationv1.ExtraValue([]string{actualPod.Spec.NodeName})
+		o.Expect(saUser.Status.UserInfo.Extra["authentication.kubernetes.io/node-name"]).To(o.Equal(expectedNode))
+
+		allowedConfigMap := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: f.Namespace.Name,
+				Name:      actualPod.Spec.NodeName,
+			},
+		}
+		disallowedConfigMap := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: f.Namespace.Name,
+				Name:      "unlikely-node",
+			},
+		}
+		disallowedMessage := fmt.Sprintf("this user running on node '%s' may not modify ConfigMap '%s' because the name does not match the node name", actualPod.Spec.NodeName, disallowedConfigMap.Name)
+
+		actualAllowedConfigMap, err := nodeScopedClient.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, allowedConfigMap, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		_, err = nodeScopedClient.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, disallowedConfigMap, metav1.CreateOptions{})
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(err.Error()).To(o.ContainSubstring(disallowedMessage))
+
+		// now create so we can see the update cases
+		actualDisallowedConfigMap, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, disallowedConfigMap, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		actualAllowedConfigMap, err = nodeScopedClient.CoreV1().ConfigMaps(f.Namespace.Name).Update(ctx, actualAllowedConfigMap, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		_, err = nodeScopedClient.CoreV1().ConfigMaps(f.Namespace.Name).Update(ctx, actualDisallowedConfigMap, metav1.UpdateOptions{})
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(err.Error()).To(o.ContainSubstring(disallowedMessage))
+
+		// ensure that if the node claim is missing from the restricted service-account user, we reject the request
+		impersonatingConfig := rest.CopyConfig(f.ClientConfig())
+		impersonatingConfig.Impersonate.UserName = saUser.Status.UserInfo.Username
+		impersonatingConfig.Impersonate.UID = saUser.Status.UserInfo.UID
+		impersonatingConfig.Impersonate.Groups = saUser.Status.UserInfo.Groups
+		impersonatingConfig.Impersonate.Extra = map[string][]string{}
+		for k, v := range saUser.Status.UserInfo.Extra {
+			if k == "authentication.kubernetes.io/node-name" {
+				continue
+			}
+			currVal := append([]string{}, v...)
+			impersonatingConfig.Impersonate.Extra[k] = currVal
+		}
+		impersonatingClient, err := kubernetes.NewForConfig(impersonatingConfig)
+		framework.ExpectNoError(err)
```

Comment:
It may be better to just do a token request directly with the cluster admin rest config.  Then you can just make requests as the SA without being bound to a node (the `make a kubeconfig with the token and confirm the kube-apiserver has the expected claims` logic above could be mostly re-used).

Reply:
Is this more than cosmetics?  Really think it's easier to reason about?

Reply:
I don't feel strongly.

To me, a regular user would have an easier time testing such a VAP using `kubectl create token` which will always result in the correct user info instead of them trying to carefully impersonate it.

Maybe it would be best to use `kubectl` directly since this is an e2e and it would line up better with docs?

Reply:
I prefer to not depend on `kubectl` in an E2E test. The daemonset on the node is not going to use `kubectl` either.

Reply:
+1 for not depending on kubectl, but I *would* probably create an actual pod-bound token with a tokenrequest directly, rather than using impersonation